README.rst

```diff
@@ -293,39 +293,41 @@ directory of your choice::
     git clone https://github.com/matrix-org/synapse.git
     cd synapse
 
-Synapse has a number of external dependencies, that are easiest
-to install using pip and a virtualenv::
+Synapse has a number of external dependencies. We maintain a fixed development
+environment using [poetry](https://python-poetry.org/). First, install poetry. We recommend
 
-    python3 -m venv ./env
-    source ./env/bin/activate
-    pip install -e ".[all,dev]"
+    pip install --user pipx
+    pipx install poetry
 
-This will run a process of downloading and installing all the needed
-dependencies into a virtual env. If any dependencies fail to install,
-try installing the failing modules individually::
+but see the `poetry installation docs <https://python-poetry.org/docs/#installation>`_
```

Comment:
My only issue with sending people to the poetry docs is that installing by `pipx` is not recommended by them, which people (beginners especially) might find confusing, especially if something isn't working. Maybe add some words that acknowledge this?

Reply:
Agreed it's worth acknowledging this.

FWIW It is mentioned further down on that link, if not their primary recommendation: see https://python-poetry.org/docs/#installing-with-pipx. Though as you say that's a point of confusion.

Their main recommendation is to download a script with curl and pipe it into Python. (That allows poetry to upgrade itself, among other things.) I'm sure it's safe---the downloaded script is small, and we trust the poetry project---but I didn't want to encourage that behaviour.